### PR TITLE
Enable DATAS GC for the Dashboard

### PR DESCRIPTION
**Pull request by eerhardt**
In .NET 8, the .NET runtime team developed a new GC mode named "DATAS (Dynamic Adaptation To Application Sizes)". This is like Server GC, but it doesn't try to consume all the memory the machine is given. It tries to dynamically adapt to how much the app needs. You can read more about this in https://maoni0.medium.com/dynamically-adapting-to-application-sizes-2d72fcb6f1ea.

Enable this GC mode for the Dashboard in order to consume less memory.

Before and after memory consumption in Task Manager after using the TestShop app and pushing around in the Dashboard (yes I know this isn't a very scientific test).

### Before
image

### After
image

(ignore the VS memory consumption. That is unrelated)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2818)

### CODE REVIEW

File: src/Aspire.Dashboard/Aspire.Dashboard.csproj [hunk 1]
```diff
@@ -3,10 +3,9 @@
   <PropertyGroup>
     <!-- This list of runtimes needs to match the set of runtimes specified in eng/workloads/workloads.csproj -->
     <RuntimeIdentifiers>win-x86;win-x64;win-arm64;linux-x64;linux-arm64;osx-x64;osx-arm64</RuntimeIdentifiers>
-    <OutputType>Exe</OutputType>
     <TargetFramework>$(NetCurrent)</TargetFramework>
-    <Nullable>enable</Nullable>
-    <ImplicitUsings>enable</ImplicitUsings>
+    <!-- Enable System.GC.DynamicAdaptationMode by default to get a GC that dynamically adapts to the application's memory use. -->
```
Comment (eerhardt) [inline src/Aspire.Dashboard/Aspire.Dashboard.csproj:7]:
```suggestion
    <!-- Enable System.GC.DynamicAdaptationMode to get a GC that dynamically adapts to the application's memory use. -->
```

**Review by adamint — APPROVED** [review]

**Review by JamesNK — APPROVED** [review]
Let's do it. If there is feedback about perf then we can re-evaluate.

**Comment by eerhardt** [comment]
>  If there is feedback about perf then we can re-evaluate.

My thinking is that feedback would be of great value to the .NET runtime team.

**Comment by mitchdenny** [comment]
Should we be considering making this the default in the Aspire templates? Given we know with a pretty high degree of certainty that Aspire-based apps are going to be containerized workloads?

**Comment by JamesNK** [comment]
IMO no. Leave the Aspire templates close to the other .NET 8 templates. Revisit in .NET 9.

File: src/Aspire.Dashboard/Aspire.Dashboard.csproj [hunk 2]
```diff
@@ -3,10 +3,9 @@
   <PropertyGroup>
     <!-- This list of runtimes needs to match the set of runtimes specified in eng/workloads/workloads.csproj -->
     <RuntimeIdentifiers>win-x86;win-x64;win-arm64;linux-x64;linux-arm64;osx-x64;osx-arm64</RuntimeIdentifiers>
-    <OutputType>Exe</OutputType>
     <TargetFramework>$(NetCurrent)</TargetFramework>
-    <Nullable>enable</Nullable>
-    <ImplicitUsings>enable</ImplicitUsings>
```
Comment (drewnoakes) [inline src/Aspire.Dashboard/Aspire.Dashboard.csproj:9]: Why were these removed?

Comment (eerhardt) [reply src/Aspire.Dashboard/Aspire.Dashboard.csproj:9]: They are unnecessary since these are the default values.

* OutputType gets defaulted to Exe by `<Project Sdk="Microsoft.NET.Sdk.Web">`
* Nullable and ImplicitUsings gets defaulted to enable in [Directory.Build.props](https://github.com/dotnet/aspire/blob/0d74a1466004eceb60ce02fbf369e72d9c399b91/Directory.Build.props#L24-L25).

**Comment by eerhardt** [comment]
> Should we be considering making this the default in the Aspire templates?

I agree with @JamesNK here. We leave the defaults the same as any other ASP.NET apps.

Note that DATAS GC mode is being considered being the default when Server GC is enabled. I looked for an issue in dotnet/runtime, but couldn't find one tracking this. @mangod9 - Do we have an issue tracking this?
